<?xml version="1.0"?>
<launch>

  <!-- LOGGING -->
  <arg name="log_internals" default="true"/>
  <arg name="log_frames" default="true"/>
  <arg name="log_latencies" default="false"/>

  <!-- CONNECTION -->
  <!-- <arg name="serverIP" default="10.1.83.130"/> -->
  <!-- 桥接模式 -->
  <arg name="serverIP" default="192.168.246.1"/>
  <!-- nat模式 -->
  <!-- 服务端以太网IPv4的地址 -->
  <!-- <arg name="clientIP" default="10.1.82.91"/> -->
  <!-- 桥接模式 -->
  <arg name="clientIP" default="192.168.246.128"/>
  <!-- nat模式 -->
  <arg name="serverType" default="multicast"/>
  <!-- 多播状态 -->
  <arg name="multicastAddress" default="239.255.42.99"/>
  <!-- 服务器命令端口 -->
  <arg name="serverCommandPort" default="1510"/>
  <!-- 服务器数据端口 -->
  <arg name="serverDataPort" default="1511"/>

  <!-- PUBLISHING -->
  <!-- 发布刚体数据 -->
  <arg name="pub_rigid_body" default="true"/> 
  <!-- 发布刚体点数据 -->
  <arg name="pub_rigid_body_marker" default="false"/>  
 
  <arg name="pub_individual_marker" default="false"/>

  <arg name="pub_pointcloud" default="false"/>

  <!-- CONFIG FILE -->
  <arg name="conf_file" default="initiate.yaml"/>

<!-- ############################### 打开节点 ############################### -->
  <node pkg="natnet_ros_cpp" type="natnet_ros_cpp" name="natnet_ros" output="screen" >    
    <rosparam file="$(find natnet_ros_cpp)/config/$(arg conf_file)"/>   
  
    <!-- LOGGING -->
    <param name="log_internals" value="$(arg log_internals)" type="bool"/>
    <param name="log_frames" value="$(arg log_frames)" type="bool"/>
    <param name="log_latencies" value="$(arg log_latencies)" type="bool"/>

    <!-- CONNECTION -->
    <param name="serverIP" value="$(arg serverIP)" type="str"/>
    <param name="clientIP" value="$(arg clientIP)" type="str"/>
    <param name="serverType" value="$(arg serverType)" type="str"/>
    <param name="multicastAddress" value="$(arg multicastAddress)" type="str"/>
    <param name="serverCommandPort" value="$(arg serverCommandPort)" type="int"/>
    <param name="serverDataPort" value="$(arg serverDataPort)" type="int"/>

    <!-- PUBLISHING -->
    <param name="pub_rigid_body" value="$(arg pub_rigid_body)" type="bool"/>
    <param name="pub_rigid_body_marker" value="$(arg pub_rigid_body_marker)" type="bool"/>
    <param name="pub_individual_marker" value="$(arg pub_individual_marker)" type="bool"/>
    <param name="pub_pointcloud" value="$(arg pub_pointcloud)" type="bool"/>
  </node>

  
</launch>
